src/Time/Units.hs

```diff
@@ -348,6 +349,28 @@ __Examples:__
 toNum :: forall (unitTo :: Rat) n (unit :: Rat) . (KnownDivRat unit unitTo, Num n)
       => Time unit -> n
 toNum = fromIntegral @Natural . floorRat . toUnit @unitTo
+{-# DEPRECATED toNum "May lead to unexpected flooring of the fractional time." #-}
+
+{- | Convert the 'Time' object to the 'Fractional' value in given units.
+
+__Examples:__
+
+>>> toFractional @Second @Rational $ hour 3
+10800 % 1
+
+>>> toFractional @Hour @Rational $ hour 3
+3 % 1
+
+>>> toFractional @Day @Rational $ hour 3
+1 % 8
+
+>>> toFractional @Day @Double $ hour 3
+0.125
+
+-}
+toFractional :: forall (unitTo :: Rat) r (unit :: Rat) . (KnownDivRat unit unitTo, Fractional r)
+             => Time unit -> r
+toFractional = fromRational . toRational . unTime . toUnit @unitTo
```

Comment:
I don't think `toFractional` should internally call `toUnit`.

This implementation assumes the input `Time unit` isn't _yet_ in the desired fractional unit.
But, if it is, then the call to `toUnit` is needless work, like in your second example:

```hs
>>> toFractional @Hour @Rational $ hour 3
3 % 1
```

IMO `toFractional` should do _only_ it what says on the tin: convert the `Time unit` to a fractional number.
If the user wants to convert the units first, then they can compose `toFractional` with `toUnit` themselves.

(PS: Yes, I know `toNum` does this, but IMO it shouldn't, and that's another reason I'm glad we're getting rid of it :sweat_smile: )

Reply:
Oh, after you make this change, please make sure the type variables are ordered according to #130